examples/pi_reduce_array.c

```diff
@@ -0,0 +1,86 @@
+#include <shmem.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <time.h>
+#include <math.h>
+
+#define NUM_POINTS 10000
+
+int
+main(int argc, char* argv[], char *envp[])
+{
+    int me, myshmem_n_pes;
+    /*
+    ** Starts/Initializes SHMEM/OpenSHMEM
+    */
+    shmem_init();
+    /*
+    ** Fetch the number or processes
+    ** Some implementations use num_pes();
+    */
+    myshmem_n_pes = shmem_n_pes();
+    /*
+    ** Assign my process ID to me
+    */
+    me = shmem_my_pe();
+
+    /*
+    ** Allocate memory on the symmetric heap
+    */
+    long long* inside = shmem_malloc(myshmem_n_pes * sizeof(long long));
+    if (inside == NULL) {
+        printf("Symmetric Heap for 'inside' not allocated\n");
+        exit(1);
+    }
+    long long* total = shmem_malloc(myshmem_n_pes * sizeof(long long));
+    if (total == NULL) {
+        printf("Symmetric Heap for 'total' not allocated\n");
+        exit(1);
+    }
+
+    srand(1+me);
+
+    long long localInside = 0, localTotal = 0;
+
+    for(localTotal = 0; localTotal < NUM_POINTS; ++localTotal) {
+        double x,y;
+        x = rand()/(double)RAND_MAX;
+        y = rand()/(double)RAND_MAX;
+
+        if(x*x + y*y < 1) {
+            ++localInside;
+        }
+    }
+
+    /*
+    ** Store each PE's local inside/total values in the inside array at an index
+    ** matching the PE's unique id
+    */
```

Comment:
I think we should remove the pi "array" example to keep things simple.  This version ends up with the correct calculation, but each PE uses only 1 index of the entire array (wasting memory).  Probably best to just use the `pi_reduce.c` version, which looks perfect!

Reply:
I agree. Had the same feeling.

Reply:
Actually, on a second thought, I was thinking whether `pi.c` and `pi_reduce.c` could be combined to have a single program with two different options using compile flags.

Reply:
@wrrobin - Did you see the pi "exercise" in Kieran's new "Getting Started Guide"?  We might not want to give the answer away to the interested reader 😉

Reply:
oh.. ok. I see..

Reply:
I just removed pi_reduce_array.c for now, if we want to combine them I can do that as well. I do believe that it may make the simple examples a little harder to follow.